# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/wasm/WasmInstance.h
# Commit: 44b0c0bed8a6
# Full Hash: 44b0c0bed8a6b4debb950e7e1c69190883c65d60
# Author: Lars T Hansen <lhansen@mozilla.com>
# Date: 2021-10-18 16:09:29
# Regressor Bug: 1727084
# File Overlap Count: 1
# Description:
#   Bug 1727084 - Memory64 - Atomic-rmw baseline operations. r=yury
#   
#   Atomic ADD, SUB, AND, OR, XOR, XCHG, CMPXCHG, as well as WAIT and
#   NOTIFY for the baseline compiler.
#   
# ==============================================================================

diff -r fb02656bb788 -r 44b0c0bed8a6 js/src/wasm/WasmInstance.h
--- a/js/src/wasm/WasmInstance.h	Mon Oct 18 09:58:15 2021 +0000
+++ b/js/src/wasm/WasmInstance.h	Mon Oct 18 09:58:15 2021 +0000
@@ -242,11 +242,18 @@
                            uint32_t srcOffset, uint32_t len, uint32_t segIndex,
                            uint32_t tableIndex);
   static int32_t elemDrop(Instance* instance, uint32_t segIndex);
-  static int32_t wait_i32(Instance* instance, uint32_t byteOffset,
-                          int32_t value, int64_t timeout);
-  static int32_t wait_i64(Instance* instance, uint32_t byteOffset,
-                          int64_t value, int64_t timeout);
-  static int32_t wake(Instance* instance, uint32_t byteOffset, int32_t count);
+  static int32_t wait_i32_m32(Instance* instance, uint32_t byteOffset,
+                              int32_t value, int64_t timeout);
+  static int32_t wait_i32_m64(Instance* instance, uint64_t byteOffset,
+                              int32_t value, int64_t timeout);
+  static int32_t wait_i64_m32(Instance* instance, uint32_t byteOffset,
+                              int64_t value, int64_t timeout);
+  static int32_t wait_i64_m64(Instance* instance, uint64_t byteOffset,
+                              int64_t value, int64_t timeout);
+  static int32_t wake_m32(Instance* instance, uint32_t byteOffset,
+                          int32_t count);
+  static int32_t wake_m64(Instance* instance, uint64_t byteOffset,
+                          int32_t count);
   static void* refFunc(Instance* instance, uint32_t funcIndex);
   static void preBarrierFiltering(Instance* instance, gc::Cell** location);
   static void postBarrier(Instance* instance, gc::Cell** location);